micropip/transaction.py

```diff
@@ -80,21 +80,15 @@ async def add_requirement(self, req: str | Requirement) -> None:
         try:
             as_req = constrain_requirement(Requirement(req), self.constrained_reqs)
         except InvalidRequirement:
-            as_req = None
-
-        if as_req:
-            if as_req.name and len(as_req.specifier):
-                return await self.add_requirement_inner(as_req)
-            if as_req.url:
-                req = as_req.url
-
-        if urlparse(req).path.endswith(".whl"):
-            # custom download location
-            wheel = WheelInfo.from_url(req)
-            check_compatible(wheel.filename)
-            return await self.add_wheel(wheel, extras=set(), specifier="")
```

Comment:
I think we need to check the parsed url after converting the str to Requirement here. We don't support non `.whl` suffixes.

Reply:
Ohh, okay. Thanks for the traceback. That makes sense.